### PR TITLE
MolecularDB flags: public/archived/targeted

### PR DESCRIPTION
**Pull request by intsco**
* Add migration scripts
* Remove moldb lists from graphql configs
* Modify MolDB GraphQL API

### CODE REVIEW

**Comment by intsco** [comment]
@LachlanStuart please verify the GraphQL API changes

File: metaspace/graphql/schemas/moldb.graphql [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  public: Boolean!
+  archived: Boolean!  # Cannot be used for dataset processing
+  targeted: Boolean!  # If true keep all annotations regardless of their FDR
+}
+
+type Query {
+  molecularDatabases(hideDeprecated: Boolean=true,
+    onlyLastVersion: Boolean=true): [MolecularDatabase!] @deprecated(reason: "Use molecularDatabasesV2 instead")
+  molecularDatabasesV2(hideArchived: Boolean=true): [MolecularDatabase!]
+}
```
Comment (intsco) [inline metaspace/graphql/schemas/moldb.graphql]: @LachlanStuart please verify the GraphQL API changes

**Comment by intsco** [comment]
@LachlanStuart BTW, as a result of these changes, the docker dev setup can be broken. Should I test it?

File: metaspace/graphql/src/modules/lookups/controller.ts [hunk 2]
```diff
@@ -85,38 +81,6 @@ const QueryResolvers: FieldResolversFor<Query, void> = {
     });
   },
 
-  async molecularDatabases(source, args, ctx) {
```
Comment (LachlanStuart) [inline metaspace/graphql/src/modules/lookups/controller.ts]: This resolver is still needed - did you intend to copy it into `moldb/controller`?

File: metaspace/graphql/src/modules/moldb/controller.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import config from "../../utils/config";
+import logger from "../../utils/logger";
+import {IResolvers} from "graphql-tools";
+import {Context} from "../../context";
+
+
+const QueryResolvers: FieldResolversFor<Query, void> = {
+  async molecularDatabasesV2(source, {hideArchived}, ctx) {
+      const molDBs = await ctx.entityManager.getRepository(MolecularDB).find();
+      return molDBs.map(molDB => {
```
Comment (LachlanStuart) [inline metaspace/graphql/src/modules/moldb/controller.ts]: This doesn't apply `hideArchived`

**Comment by LachlanStuart** [comment]
@intsco 
> BTW, as a result of these changes, the docker dev setup can be broken. Should I test it?

The changes I see in this PR wouldn't have broken it. In fact removing `sm-mol-db` from the `docker-compose.yml` would have actually fixed one of the issues

File: metaspace/graphql/src/modules/lookups/controller.ts (hunk 2 above)
Comment (intsco) [reply metaspace/graphql/src/modules/lookups/controller.ts]: Yes, I planned to reimplement it using the `molecularDatabasesV2` but now I'm now sure anymore whether it's the best way to go.

**Pull request merged**